system/lib/libc/sigaction.c

```diff
@@ -0,0 +1,22 @@
+#include <stdio.h>
+#include <signal.h>
+#include <errno.h>
+#include "libc.h"
+
+struct sigaction __sig_actions[_NSIG];
+
+int __sigaction(int sig, const struct sigaction *restrict sa, struct sigaction *restrict old) {
+  if (sig < 0 || sig >= _NSIG) {
+    errno = EINVAL;
+    return -1;
+  }
+
+  if (old) {
+    *old = __sig_actions[sig];
+  }
+
+  __sig_actions[sig] = *sa;
+  return 0;
+}
+
+weak_alias(__sigaction, sigaction);
```

Comment:
why does this one use a weak alias, but the others do not?

Reply:
musl internally relies on `__sigaction` (the internal version).  For the other symbols it does not and we don't need them.

Reply:
See https://github.com/emscripten-core/emscripten/blob/0fe9b0b2f169a565fb5f419e696ddaeb710de8ef/system/lib/libc/musl/src/signal/signal.c#L5-L13